src/_main_/utils/translation/__init__.py

```diff
@@ -0,0 +1,71 @@
+from _main_.utils.translation.google_translate import translate_json_with_google
+
+JSON_EXCLUDE_KEYS = {
+    'id', 'pk', 'file', 'media', 'date'
+}
+
+
+class JsonTranslator:
+    def __init__(self, dict_to_translate, exclude_keys={}):
+        self.exclude_keys = exclude_keys
+        self.sep = '.'
+        self._flattened, self._excluded = self.flatten_dict_for_translation(dict_to_translate)
+
+    def flatten_dict_for_translation(self, dict_to_translate):
+        stack = [((), dict_to_translate)]
+        flattened_dict_for_keys_to_include = {}
+        flattened_dict_for_keys_to_exclude = {}
+        
+        while stack:
+            parent_key, current = stack.pop()
+            for k, v in current.items():
+                new_key = self.sep.join((parent_key, k)) if parent_key else k
+                if self._should_exclude(k, v):
+                    flattened_dict_for_keys_to_exclude[new_key] = v
+                else:
+                    if isinstance(v, dict):
+                        stack.append((new_key, v))
+                    else:
+                        flattened_dict_for_keys_to_include[new_key] = v
+        return flattened_dict_for_keys_to_include, flattened_dict_for_keys_to_exclude
+
+    def _should_exclude(self, key, _value):
+        #TODO: 1. add more logic here to exclude keys called file, media, etc
+        #TODO: 2. we can also add logic to say if _value starts with https://s3.ama...... then return true
+        #      this sill ensure that files are always excluded
+        res = (
+            (key in self.exclude_keys) or 
+            (key in JSON_EXCLUDE_KEYS) or 
+            not (isinstance(_value, str) or isinstance(_value, dict))
```

Comment:
we want to exclude any key in the exclusion lists and also if the value is not a string or dict.